src/qml/components/Avatar.qml

```diff
@@ -1,38 +1,71 @@
 import QtQuick 2.12
+import QtGraphicalEffects 1.0
 
 import "../theme"
 import "./CommonHelpers"
 
-TextInCircle {
+Item {
     property string nickname
+    property string avatarUrl
 
-    color: Theme.avatarBgColor // intToHexColor(hashCode(nickname))
-    diameter: Theme.avatarHeight
-    content: nickname.replace("_", "").substring(0, 2).toUpperCase()
-    pointSize: UiHelper.fixFontSz(0.4 * diameter)
-
-    // Hash any string into an integer value
-    // Then we'll use the int and convert to hex.
-    function hashCode(str) {
-        var hash = 0;
-        for (var i = 0; i < str.length; i++) {
-            hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    width: Theme.avatarHeight
+    height: width
+
+    TextInCircle {
+        color: Theme.avatarBgColor // intToHexColor(hashCode(nickname))
+        anchors.centerIn: parent
+        diameter: parent.height
```

Comment:
This property isn't externally visible more, pages that use avatar diameter are broken now. Please check other properties that you moved from component scope and test pages with avatars

Reply:
Created alias diameter: textInCircle.diameter

Reply:
and content and pointSize